src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,140 @@
+
+        Node(Node<T> prev, T item, Node<T> next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<T>(null, value, null);
+        if (first == null) {
+            first = newNode;
+        } else {
+            newNode.prev = last;
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (size == index) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        Node<T> currentNode = findNode(index);
+        Node<T> newNode = new Node<T>(currentNode.prev, value, currentNode);
+        if (currentNode.prev == null) {
+            first = newNode;
+        } else {
+            newNode.prev.next = newNode;
+        }
+        newNode.next.prev = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return findNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> currentNode = findNode(index);
+        T returnValue = currentNode.item;
+        currentNode.item = value;
+        return returnValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> currentNode = findNode(index);
+        unlink(currentNode);
+        return currentNode.item;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> currentNode = first;
+        for (int i = 0; i < size; i++) {
+            if (object == currentNode.item || currentNode.item != null
+                    && currentNode.item.equals(object)) {
+                unlink(currentNode);
+                return true;
+            }
+            currentNode = currentNode.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void unlink(Node node) {
+        if (node.next != null) {
+            node.next.prev = node.prev;
+        } else {
+            last = node.prev;
+        }
+        if (node.prev != null) {
+            node.prev.next = node.next;
+        } else {
+            first = node.next;
+        }
+        size--;
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Incorect index: " + index);
+        }
+    }
+
+    private Node<T> findNode(int index) {
+        if (index > size / LIST_DIVIDER) {
```

Comment:
And this is a redundant constant because it is quite obvious without any additional names that we need to divide a size into 2.